### PR TITLE
Trapi 1.1

### PR DESCRIPTION
**Pull request by YaphetKG**

### CODE REVIEW

File: PLATER/models/shared.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        for q_node_id in q_nodes:
+            q_node = q_nodes[q_node_id]
+            if q_node.id:
+                has_bound_node = True
+                break
+        assert has_bound_node, "Query graph should contain at least one bound node."
+        return v
+
+    @validator("message")
+    def validate_broken_edge(cls, v):
```
Comment (patrickkwang) [inline PLATER/models/shared.py:21]: This is handy! I'd like to implement some of this stuff in reasoner-pydantic.

File: PLATER/requirements.txt [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 pytest-asyncio==0.14.0
 starlette==0.13.6
 uvicorn==0.11.7
-git+git://github.com/patrickkwang/biolink-model-toolkit@master#egg=bmt
 git+https://github.com/ranking-agent/reasoner.git@main#egg=reasoner
-git+https://github.com/TranslatorSRI/reasoner-pydantic#egg=reasoner-pydantic
-git+https://github.com/patrickkwang/fastapi#egg=fastapi
+git+https://github.com/TranslatorSRI/reasoner-pydantic@v1.1.0#egg=reasoner-pydantic
 httpx
+pytest-httpx
```
Comment (patrickkwang) [inline PLATER/requirements.txt:12]: We developed a similar, but perhaps more flexible, tool for testing Strider: https://github.com/patrickkwang/asgiar

File: PLATER/services/app_trapi_1_0.py [hunk 3]
```diff
@@ -1,80 +1,43 @@
...
-) -> Message:
+):
     """Handle TRAPI request."""
     request_json = request.dict()
     question = Question(request_json["message"])
     response = await question.answer(graph_interface)
-    request_json.update({'message': response})
-    return request_json
+    request_json.update({"message": response})
+    return JSONResponse(request_json)
```
Comment (patrickkwang) [inline PLATER/services/app_trapi_1_0.py:37]: I'm curious why this is necessary. FastAPI typically generates a JSON response by default.

File: PLATER/services/util/graph_adapter.py [hunk 4]
```diff
@@ -227,17 +232,18 @@ def get_schema(self):
                 structured = self.convert_to_dict(result)
                 self.schema_raw_result = structured
                 schema_bag = {}
-                # permituate source labels and target labels array
+                # permtuate source labels and target labels array
```
Comment (patrickkwang) [inline PLATER/services/util/graph_adapter.py:235]: Permute?

**Review by patrickkwang — CHANGES_REQUESTED** [review]
Looks good. Minor comments.

File: PLATER/services/app_trapi_1_0.py (hunk 3 above)
Comment (YaphetKG) [reply PLATER/services/app_trapi_1_0.py:37]: This was to allow backward compatibility, i.e 1.0 responses I was running into validation errors as the default fast api response would pass through the 1.1 validation. Will reverfy with the latest,

File: PLATER/requirements.txt (hunk 2 above)
Comment (YaphetKG) [reply PLATER/requirements.txt:12]: sounds good will checkit out and use that

File: PLATER/examples/reasoner-trapi-1.1.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+            "categories":[
+               "biolink:Disease"
+            ]
+         }
+      },
+      "edges":{
+         "e0":{
+            "subject":"n0",
+            "object":"n1",
+            "predicates":"biolink:related_to"
```
[the remark this answers is not in the document]
Comment (YaphetKG) [reply PLATER/examples/reasoner-trapi-1.1.json]: predicates is now array in exampels.

File: PLATER/requirements.txt (hunk 2 above)
Comment (YaphetKG) [reply PLATER/requirements.txt]: Hi @patrickkwang looked into asgiar, but didn't find an easy work around with out refactoring the tests. maybe we can differ this to an issue and track it seperately (?). There are some standing issues that I think we could prioritize before this refactor.

File: PLATER/services/app_trapi_1_0.py (hunk 3 above)
Comment (YaphetKG) [reply PLATER/services/app_trapi_1_0.py:37]: Latest Transplier has some changes that require some refactoring in plater. For this PR I think we should use a commit version of transpliler in our requirements and Work on that refactor on the next set of things.

**Review by patrickkwang — APPROVED** [review]

**Pull request merged**